library_generation/README.md

```diff
@@ -0,0 +1,112 @@
+# Generate GAPIC Client Library without post-processing
+
+The script, `generate_library.sh`, allows you to generate a GAPIC client library from proto files.
+
+## Environment
+
+Use Linux environment and install java runtime environment (8 or above).
+
+## Prerequisite
+In order to generate a GAPIC library, you need to pull `google/` from [googleapis](https://github.com/googleapis/googleapis)
+and put it into the directory containing `generate_library.sh` since protos in `google/` 
```

Comment:
We have to copy the whole `google/` folder? Or maybe only the common protos?

Reply:
Some protos reference `google/api`, `google/iam`, `google/type`. I'm not sure whether we have a exhausted list of protos such that only those protos can be referenced. So copy `google` folder is a safe option.

Reply:
Not part of this PR, but I think we should come up with an exhaustive list before we switch to use it in our generation process. As `google/` folder contains protos from all services, which is a lot more than what we need, and it could easily confuse people in the future.

Reply:
We can do that, but do we have a guidance that a proto should not reference a proto outside of a list? If not, the list is likely changing and we are back to copy `google`.

Reply:
Theoretically, the service team can reference protos from anywhere, but cross service proto depedencies are not allowed per https://google.aip.dev/213, so the service protos should only reference common protos from a few selected folders. I think we may have two services that have cross service proto dependencies but all other services should obey it.

 Now regarding common protos, we don't have a guidance on where they should be and it's possible that new common protos are added to new locations. But if it happens, I think it would be better to explicitly add it, similar to `explicitly declare a new dependency` vs `getting a new dependency transitively from a existing dependency` in Maven's world.

We can discuss more in next project meeting.